target_iceberg/sinks.py

```diff
@@ -10,48 +10,97 @@
 from pyiceberg.catalog import load_catalog
 from singer_sdk.helpers._flattening import flatten_record, flatten_schema
 from singer_sdk.sinks import BatchSink
+from singer_sdk.exceptions import ConfigValidationError
 
-from target_iceberg.utils import create_pyarrow_table, flatten_schema_to_pyarrow_schema, process_config_replace
+from target_iceberg.utils import (create_pyarrow_table, flatten_schema_to_pyarrow_schema,
+                                  process_config_replace, clean_split, to_snake_case)
 
 
+SYNCED_COLUMN_NAME = "synced_ms"
+
 class IcebergSink(BatchSink):
-    spark = None
     def __init__(self, target, schema, stream_name, key_properties) -> None:
         super().__init__(
             target=target,
             schema=schema,
             stream_name=stream_name,
             key_properties=key_properties,
         )
-        snake_case_stream_name = IcebergSink.to_snake_case(self.stream_name)
+        self.flatten_max_level = self.config.get("max_flatten_level", 0)
+        self.skip_add_synced_field = self.config.get("skip_add_synced_field", False)
+        self.data_buffer = None
+        self.start_time = datetime.utcnow()
+
+        self.validate_config()
+
+    def validate_config(self) -> None:
+        # Check column renames
+        missing_keys = set(self.column_renames.keys()) - set(self.flatten_schema.get("properties", {}).keys())
+        if missing_keys:
+            raise ConfigValidationError(f"Some columns marked from rename do not exist in schema: {missing_keys}")
+
+        # Check primary key if upsert is set
+        if self.upsert_data and not self.primary_key:
+            raise ConfigValidationError("Upsert is set, but no primary key defined.")
```

Comment:
Let's reuse the `process_json_config` from this [PR](https://github.com/Automattic/target-iceberg/pull/14)